### DISCUSSION TITLE
Switching GTK theme prevented by GTK_THEME ENV var?

### DISCUSSION THREAD

**Discussion by eterps** [Manjaro-Sway/manjaro-sway#631] [Q&A] [answered]
Switching GTK themes doesn't work (f.i. using [nwg-look](https://github.com/nwg-piotr/nwg-look#nwg-look)).

It seems this is because the environment variable `GTK_THEME` overrides every attempt to do so.
This variable is set in `/etc/skel/.profile`/`~/.profile`.

What is considered the 'official' way in changing a GTK theme in Manjaro-Sway? Only by overriding this variable in `~/.profile`?

**Comment by boredland** [comment]
Well, we're in a dead-lock there of sorts: gtk4 basically disables theme-switching for adwaita-enabled applications (so at least all official gnome applications and many more). For now it only works with said environment variable. But: I am experimenting with just setting gtk4 to dark-mode and not theme gtk4 applications. Which might be nice enough.

**Reply by eterps** [reply]
Thank you for your reply. I was not aware of this:

> gtk4 basically disables theme-switching for adwaita-enabled applications

Can you provide a bit more context?

**Reply by boredland** [reply]
this being said: you can disable that variable and try it on you own (good applications to test are pamac or gnome-firmware). If you want to persist unsetting that variable:

```
echo "unset GTK_THEME" > .config/profile.d/disable-gtk-theme
```

**Reply by boredland** [reply]
You can try the dark-mode of adwaita like so:

```
ADW_DEBUG_COLOR_SCHEME=prefer-dark pamac-manager
```

matcha-dark-theme:
![image](https://github.com/manjaro-sway/manjaro-sway/assets/4662748/2cb839cb-5f5a-46b9-9460-233da2f47732)

adwaita:
![swappy-20231126_125526](https://github.com/manjaro-sway/manjaro-sway/assets/4662748/36fac415-558d-4012-80ef-c491937f05d2)

I am still not sure what I could do in the future, as the GTK_THEME variable will be dropped.

**Reply by boredland** [reply]
People seem to maintain accent colors in a css file. Which looks like a nightmare to maintain:
https://forum.manjaro.org/t/how-to-fix-appearance-of-gtk4-apps-with-dark-themes-on-xfce/142790

**Reply by boredland** [reply]
If you find a way that works for both, gtk3 and gtk4, I am all ears!

**Reply by boredland** [reply]
I think just running `/usr/bin/cp -rf /usr/share/themes/$(gsettings get org.gnome.desktop.interface gtk-theme | sed -e "s#'##g")/gtk-4.0/{assets,gtk.css,gtk-dark.css} $HOME/.config/gtk-4.0/` when sway reloads might be viable together with the adw color-scheme preference.

**Reply by boredland** [reply, marked as answer]
Sooo, I think that might actually work. `manjaro-sway-settings-git` has it. Could you please:

1. remove overrides from `.config/profile.d`
2. install `manjaro-sway-settings-git`
3. run `skel`
4. restart your session (reload is not enough due to changes in `.profile`)

And than check if your theme is being applied?

**Reply by eterps** [reply]
Thank you, switching GTK themes works now. It seems using nwg-look does not persist the changed GTK settings, but that's a seperate issue I will figure out.

**Reply by boredland** [reply]
Well, do they persist for gtk3? Because I think nwg-look is only for gtk3... My assumption would be, that it will work fine for gtk3 apps and not do anything for gtk4 apps.

Ah, well: https://github.com/nwg-piotr/nwg-look/issues/22
